api/workAllocation/util.ts

```diff
@@ -301,9 +301,13 @@ export function getRoleCategory(roleAssignments: RoleAssignment[], caseWorkerApi
   return roleAssignment ? roleAssignment.roleCategory : null;
 }
 
-export function getUserRoleCategory(roleAssignments: RoleAssignment[], user: StaffProfile): string {
+export function getUserRoleCategory(roleAssignments: RoleAssignment[], user: StaffProfile, services: string[]): string {
   // TODO: Will need to be updated
-  const roleAssignment = roleAssignments.find((roleAssign) => roleAssign.actorId === user.id);
+  const roleAssignment = roleAssignments.find((roleAssign) =>
+    roleAssign.actorId === user.id &&
+    // added line below to stop irrelevant role setting role category
+    (!roleAssign.attributes?.jurisdiction || services.includes(roleAssign.attributes.jurisdiction))
```

Comment:
Should also check that the roleAssign has a category - don't want to end up returning one without a category
e.g roleAssign.roleCategory && (!roleAssign.attributes?.jurisdiction || services.includes(roleAssign.attributes.jurisdiction))